### ISSUE TITLE
Arguments to path.join must be strings

### ISSUE THREAD

**Issue by 0vermind** [avoidwork/turtle.io#346]
I get
>Arguments to path.join must be strings

on fedora 24
installation with no issue and requests are logged normally.
When I start the default seer.js
>node sever.js

server.start({
    default : "www.mycompany.com",
    uid     : 0,
    port    : 9999,
    root    : "/var/www",
    vhosts  : {
	"www.mycompany.com"      : "mycompany.com",
	"www.mycompany.com"  : "mycompany.com",
	"blog.mycompany.com" : "blog.mycompany.com"

**Comment by avoidwork** [comment]
Hi,

That is an odd message to get. I'll test tonight with the sample settings above.

**Comment by avoidwork** [comment]
I can't reproduce this error without more details; can you provide the stack, or any more info? I used the config you provided, which is invalid paths on my laptop.

```console
Last login: Wed Aug  5 06:28:25 on console
jason@avoidwork:~$ cd Projects/
jason@avoidwork:~/Projects$ mkdir test
mkdir: test: File exists
jason@avoidwork:~/Projects$ mkdir turtle-test
jason@avoidwork:~/Projects$ cd turtle-test/
jason@avoidwork:~/Projects/turtle-test$ npm install turtle.io
 
> dtrace-provider@0.4.0 install /Users/jason/Projects/turtle-test/node_modules/turtle.io/node_modules/dtrace-provider
> node scripts/install.js

|
> kerberos@0.0.12 install /Users/jason/Projects/turtle-test/node_modules/turtle.io/node_modules/keigai/node_modules/mongodb/node_modules/mongodb-core/node_modules/kerberos
> (node-gyp rebuild 2> builderror.log) || (exit 0)

  CXX(target) Release/obj.target/kerberos/lib/kerberos.o
  CXX(target) Release/obj.target/kerberos/lib/worker.o
  CC(target) Release/obj.target/kerberos/lib/kerberosgss.o
  CC(target) Release/obj.target/kerberos/lib/base64.o
  CXX(target) Release/obj.target/kerberos/lib/kerberos_context.o
  SOLINK_MODULE(target) Release/kerberos.node
  SOLINK_MODULE(target) Release/kerberos.node: Finished
turtle.io@4.0.19 node_modules/turtle.io
├── precise@1.0.1
├── mime@1.3.4
├── murmurhash3js@3.0.1
├── moment@2.10.6
├── dtrace-provider@0.4.0 (nan@1.5.3)
└── keigai@1.3.19 (localStorage@1.0.3, btoa@1.1.2, es6-promise@2.0.1, mongodb@2.0.39)
jason@avoidwork:~/Projects/turtle-test$ nano server.js
jason@avoidwork:~/Projects/turtle-test$ node server.js 
localhost:9999 - - [5/Aug/2015:06:38:41 -0400] "GET / HTTP/1.1" 404 145 "-" "GraphicalHttpClient 1.0.8 (Macintosh; Mac OS X 10.11.0; en_CA)"
```

**Comment by avoidwork** [comment]
If that's all there is to your script, could you provide more env details such that I can try to recreate with a vm; what version of node? what partition type? etc.

**Comment by 0vermind** [comment]
Hi @avoidwork thanks for the reply, I solved the issue using the standard server.js and reinstalling turtle.io
 Now I'm happily experimenting with turtle.io. But is you need some info, please give the command to get them :baby:

**Comment by avoidwork** [comment]
Shall I reopen this? I'd like to understand how it's failing for you, because maybe it needs to be more robust.

**Comment by 0vermind** [comment]
thanks, I will give you every info you need but I need to know how.

**Comment by avoidwork** [comment]
The script you're running, and the env details (granular is good), & of course the request (URL, method, querystring/headers if applicable) should be good enough to try to reproduce the error. Usually errors are really easy to solve once I can identify them.

**Comment by avoidwork** [comment]
Could you paste the server.js content into a comment, ideally wrapped with "```" for formatting ;)

**Comment by 0vermind** [comment]
I start this with node server.js
 
```js
var TurtleIO = require("turtle.io"),
    server = new TurtleIO();

server.start({
    default : "do8.xxx.net",
    uid     : 100,
    port    : 10000,
    root    : "/var/www/xxx.net/default",
    vhosts  : {
	"live.xxx.net"      : "live.xxx.net",
	"www.mycompany.com"  : "mycompany.com",
	"blog.mycompany.com" : "blog.mycompany.com"
    }
});
```

Here the env variables.
```bash
[root@do8 test-turtle]# env 
XDG_SESSION_ID=53
HOSTNAME=do8.xxx.net
SELINUX_ROLE_REQUESTED=
SHELL=/bin/bash
TERM=xterm-256color
HISTSIZE=1000
SSH_CLIENT=127.0.0.1 39785 22
SELINUX_USE_CURRENT_RANGE=
SSH_TTY=/dev/pts/0
USER=root
LS_COLORS=rs=0:di=38;5;33:ln=38;5;51:mh=00:pi=40;38;5;11:so=38;5;13:do=38;5;5:bd=48;5;232;38;5;11:cd=48;5;232;38;5;3:or=48;5;232;38;5;9:mi=01;05;37;41:su=48;5;196;38;5;15:sg=48;5;11;38;5;16:ca=48;5;196;38;5;226:tw=48;5;10;38;5;16:ow=48;5;10;38;5;21:st=48;5;21;38;5;15:ex=38;5;40:*.tar=38;5;9:*.tgz=38;5;9:*.arc=38;5;9:*.arj=38;5;9:*.taz=38;5;9:*.lha=38;5;9:*.lz4=38;5;9:*.lzh=38;5;9:*.lzma=38;5;9:*.tlz=38;5;9:*.txz=38;5;9:*.tzo=38;5;9:*.t7z=38;5;9:*.zip=38;5;9:*.z=38;5;9:*.Z=38;5;9:*.dz=38;5;9:*.gz=38;5;9:*.lrz=38;5;9:*.lz=38;5;9:*.lzo=38;5;9:*.xz=38;5;9:*.bz2=38;5;9:*.bz=38;5;9:*.tbz=38;5;9:*.tbz2=38;5;9:*.tz=38;5;9:*.deb=38;5;9:*.rpm=38;5;9:*.jar=38;5;9:*.war=38;5;9:*.ear=38;5;9:*.sar=38;5;9:*.rar=38;5;9:*.alz=38;5;9:*.ace=38;5;9:*.zoo=38;5;9:*.cpio=38;5;9:*.7z=38;5;9:*.rz=38;5;9:*.cab=38;5;9:*.jpg=38;5;13:*.jpeg=38;5;13:*.gif=38;5;13:*.bmp=38;5;13:*.pbm=38;5;13:*.pgm=38;5;13:*.ppm=38;5;13:*.tga=38;5;13:*.xbm=38;5;13:*.xpm=38;5;13:*.tif=38;5;13:*.tiff=38;5;13:*.png=38;5;13:*.svg=38;5;13:*.svgz=38;5;13:*.mng=38;5;13:*.pcx=38;5;13:*.mov=38;5;13:*.mpg=38;5;13:*.mpeg=38;5;13:*.m2v=38;5;13:*.mkv=38;5;13:*.webm=38;5;13:*.ogm=38;5;13:*.mp4=38;5;13:*.m4v=38;5;13:*.mp4v=38;5;13:*.vob=38;5;13:*.qt=38;5;13:*.nuv=38;5;13:*.wmv=38;5;13:*.asf=38;5;13:*.rm=38;5;13:*.rmvb=38;5;13:*.flc=38;5;13:*.avi=38;5;13:*.fli=38;5;13:*.flv=38;5;13:*.gl=38;5;13:*.dl=38;5;13:*.xcf=38;5;13:*.xwd=38;5;13:*.yuv=38;5;13:*.cgm=38;5;13:*.emf=38;5;13:*.ogv=38;5;13:*.ogx=38;5;13:*.aac=38;5;45:*.au=38;5;45:*.flac=38;5;45:*.m4a=38;5;45:*.mid=38;5;45:*.midi=38;5;45:*.mka=38;5;45:*.mp3=38;5;45:*.mpc=38;5;45:*.ogg=38;5;45:*.ra=38;5;45:*.wav=38;5;45:*.oga=38;5;45:*.opus=38;5;45:*.spx=38;5;45:*.xspf=38;5;45:
MC_TMPDIR=/tmp/mc-root
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/root/bin
MAIL=/var/spool/mail/root
PWD=/home/js/test-turtle
LANG=en_GB.UTF-8
SELINUX_LEVEL_REQUESTED=
HISTCONTROL=ignorespace
HOME=/root
SHLVL=2
MC_SID=31505
LOGNAME=root
SSH_CONNECTION=127.0.0.1 39785 46.101.198.244 22
LESSOPEN=||/usr/bin/lesspipe.sh %s
XDG_RUNTIME_DIR=/run/user/0
OLDPWD=/home/js
_=/usr/bin/env

```

I censored sensible data like IPs.

**Comment by avoidwork** [comment]
Sorry, I didn't mean 'env variables' as turtle.io purposefully ignores them; I meant the machine/host env. I know it's fedora24, but 32bit or 64? what file system are you running? Providing the env variables will help to recreate the condition.

**Comment by avoidwork** [comment]
I'm ignorant on this one, but is uid 100 the default www group, or is it your user? I'm usually testing against a debian flavor.

**Comment by 0vermind** [comment]
Sorry but I'm not so good with linux, so If you could specify the command I need to get info you need it will be helpful.
It's a fedora 24 64bit intel with npm installed as root on a ext4 partition.

**Comment by avoidwork** [comment]
what version of node or io are you using?

**Comment by 0vermind** [comment]
The one in YUM : v0.10.36

**Comment by avoidwork** [comment]
ok thanks.

**Comment by avoidwork** [comment]
Sorry the delay, my wife had a birthday and I'm in a bit of a 'crunch time' scenario with work. I hope to have an answer sometime this weekend.

**Comment by avoidwork** [comment]
Hi,

I just fixed this issue at work on an unrelated project. Can you give me a console.log of `process.env` ... one of the values is either undefined or not a string.

**Comment by avoidwork** [comment]
Looks like the github grammar parser isn't so smrt.

@0vermind Can you start a node repl and do the following, I'm thinking your `MC_TMP` may be tripping up `os`:

```console
$ node
> var os = require('os';
> os.tmpdir();
```

**Comment by 0vermind** [comment]
Hi, sorry for the delay,
here the output:

> os.tmpdir();
'/tmp'
>

**Comment by avoidwork** [comment]
hmm, thanks.

**Comment by avoidwork** [comment]
fixed by design with 6.0
